infra/migrations/1715808549919_add-search-box.js

```diff
@@ -0,0 +1,56 @@
+exports.up = async (pgm) => {
+  await pgm.createTable('searches', {
+    id: {
+      type: 'uuid',
+      primaryKey: true,
+      notNull: true,
+      default: pgm.func('gen_random_uuid()'),
+    },
+    search_term: {
+      type: 'varchar(255)',
+      notNull: true,
+    },
+    search_count: {
+      type: 'integer',
+      notNull: true,
+      default: 0,
+    },
+    created_at: {
+      type: 'timestamp with time zone',
+      notNull: true,
+      default: pgm.func("(now() at time zone 'utc')"),
+    },
+    updated_at: {
+      type: 'timestamp with time zone',
+      notNull: true,
+      default: pgm.func("(now() at time zone 'utc')"),
+    },
+  });
+
+  await pgm.createConstraint('searches', 'search_term_unique', {
+    unique: 'search_term',
+  });
+
+  await pgm.db.query(`
+    CREATE INDEX idx_search ON contents USING GIN(to_tsvector('portuguese', title));
+  `);
+
+  await pgm.db.query(`
```

Comment:
Não seria suficiente criar um índice ao invés de uma nova coluna?

Caso realmente precise da nova coluna, falta a atualização dos conteúdos já existentes, certo?

Reply:
@aprendendofelipe esse INDEX ele ajuda na hora de procurar um termo, sem ele iria levar segundos para encontrar algo.

A nova coluna chamada "ts" ela vai ser gerada automaticamente ao ser criada e toda vez que um titulo ou body for atualizado, os termos dela será atualizado para criar o peso dentre o que está no body e no titulo.

Reply:
Pela sua resposta, desconfio que você não interpretou corretamente minhas perguntas.

Quero saber se precisamos realmente da coluna, já que um índice parece ser suficiente e mais interessante.

Caso a coluna realmente seja necessária, falta no PR algo para atualizar os conteúdos que já existem.

Reply:
@aprendendofelipe

A coluna ts que está sendo criada é para relevancia na busca fazendo um peso entre body e title; Ao ser criada é gerada automaticamente caso exista dados.

O INDEX sendo criado vai ajudar na hora de buscar um termo.